src/apps/eos/actions/layout.py

```diff
@@ -0,0 +1,404 @@
+    await require_confirm(ctx, text, ButtonRequestType.ConfirmOutput)
+
+
+async def confirm_action_linkauth(ctx, msg: EosActionLinkAuth):
+    await ctx.call(
+        ButtonRequest(code=ButtonRequestType.ConfirmOutput), MessageType.ButtonAck
+    )
+
+    text = "Link Auth"
+    fields = []
+    fields.append("Account:")
+    fields.append(helpers.eos_name_to_string(msg.account))
+    fields.append("Code:")
+    fields.append(helpers.eos_name_to_string(msg.code))
+    fields.append("Type:")
+    fields.append(helpers.eos_name_to_string(msg.type))
+    fields.append("Requirement:")
+    fields.append(helpers.eos_name_to_string(msg.requirement))
+
+    pages = list(chunks(fields, FOUR_FIELDS_PER_PAGE))
+    paginator = paginate(show_lines_page, len(pages), FIRST_PAGE, pages, text)
+
+    await ctx.wait(paginator)
+
+
+async def confirm_action_unlinkauth(ctx, msg: EosActionUnlinkAuth):
+    await ctx.call(
+        ButtonRequest(code=ButtonRequestType.ConfirmOutput), MessageType.ButtonAck
+    )
+
+    text = "Unlink Auth"
+    fields = []
+    fields.append("Account:")
+    fields.append(helpers.eos_name_to_string(msg.account))
+    fields.append("Code:")
+    fields.append(helpers.eos_name_to_string(msg.code))
+    fields.append("Type:")
+    fields.append(helpers.eos_name_to_string(msg.type))
+
+    pages = list(chunks(fields, FOUR_FIELDS_PER_PAGE))
+    paginator = paginate(show_lines_page, len(pages), FIRST_PAGE, pages, text)
+
+    await ctx.wait(paginator)
+
+
+async def confirm_action_newaccount(ctx, msg: EosActionNewAccount):
+    await ctx.call(
+        ButtonRequest(code=ButtonRequestType.ConfirmOutput), MessageType.ButtonAck
+    )
+
+    text = "New Account"
+    fields = []
+    fields.append("Creator:")
+    fields.append(helpers.eos_name_to_string(msg.creator))
+    fields.append("Name:")
+    fields.append(helpers.eos_name_to_string(msg.name))
+    fields += authorization_fields(msg.owner)
+    fields += authorization_fields(msg.active)
+
+    pages = list(chunks(fields, FOUR_FIELDS_PER_PAGE))
+    paginator = paginate(show_lines_page, len(pages), FIRST_PAGE, pages, text)
+
+    await ctx.wait(paginator)
+
+
+async def confirm_action_unknown(ctx, action, checksum):
+    await ctx.call(
+        ButtonRequest(code=ButtonRequestType.ConfirmOutput), MessageType.ButtonAck
+    )
+    text = "Arbitrary data"
+    fields = []
+    fields.append("Contract:")
+    fields.append(helpers.eos_name_to_string(action.account))
+    fields.append("Action Name:")
+    fields.append(helpers.eos_name_to_string(action.name))
+
+    fields.append("Checksum: ")
+    fields += split_data(hexlify(checksum).decode("ascii"))
+
+    pages = list(chunks(fields, FIVE_FIELDS_PER_PAGE))
+    paginator = paginate(show_lines_page, len(pages), FIRST_PAGE, pages, text)
+
+    await ctx.wait(paginator)
+
+
+@ui.layout
+async def show_lines_page(page: int, page_count: int, pages: list, header: str):
+    lines = ["%s" % field for field in pages[page]]
```

Comment:
why not use `pages[page]` directly?

Reply:
We missed conversion for some field, that was kind of a "fix". But we have fixed this conversion and currently it uses pages directly. So this is resolved.